openedx/frontend-app-authoring
# Fix warnings in Problem Editor [FC-0062]

## Description

When you open a blank capa problem XBlock for editing, a ton of `propTypes` validation warnings appear in the console. This PR fixes those warnings and converts the validation to TypeScript (which is better than propTypes because TypeScript gets checked during CI, but propTypes warnings can easily be ignored).

## Supporting information

Part of my work toward https://github.com/openedx/frontend-app-course-authoring/issues/1086

Private ref: [FAL-3825](https://tasks.opencraft.com/browse/FAL-3825)

## Testing instructions

1. In a course, create a **blank** capa problem component. Edit it.
2. Without this fix, you'll see a ton of warnings in the JS console. With this fix in place, you won't.
3. Verify that the editor still works and can save changes to the component.

(Focus on testing from a course more so than from a library - at this point it's more important we don't break the course workflow)

## Full list of warnings fixed
```
Warning: Failed prop type: The prop `onClose` is marked as required in `ProblemEditor`, but its value is `null`.
Warning: Failed prop type: The prop `problemType` is marked as required in `ProblemEditor`, but its value is `null`.
Warning: Failed prop type: The prop `blockValue` is marked as required in `ProblemEditor`, but its value is `null`.
Warning: Failed prop type: Invalid prop `blockValue.status` of type `number` supplied to `ProblemEditor`, expected `object`.
Warning: Failed prop type: The prop `onClose` is marked as required in `SelectTypeModal`, but its value is `null`.
Warning: Failed prop type: The prop `alt` is marked as required in `IconButton`, but its value is `undefined`.
Warning: Each child in a list should have a unique "key" prop.
Warning: Failed prop type: The prop `name` is marked as required in `ForwardRef(_c)`, but its value is `undefined`.
Warning: Failed prop type: The prop `name` is marked as required in `FormRadioSet`, but its value is `undefined`.
Warning: Failed prop type: The prop `name` is marked as required in `FormRadioSetContextProvider`, but its value is `undefined`.
```
and when clicking on "Advanced problem types", the following warnings:
```
Failed prop type: The prop `alt` is marked as required in `IconButton`, but its value is `undefined`.
Failed prop type: The prop `id` is marked as required in `Tooltip`, but its value is `undefined`.
Each child in a list should have a unique "key" prop. (`ActionRow`)
```
